### DISCUSSION TITLE
@wagmi/cli - Generate ABIs from a directory of .json files

### DISCUSSION THREAD

**Discussion by gg2001** [wevm/wagmi#3019] [Ideas]
Would be nice to be able to generate ABIs from a directory of .json files, rather than having to use the foundry plugin and specify the artifacts directory

**Comment by tmm** [comment]
You can read them from the file system using an [async config](https://wagmi.sh/cli/configuration/configuring-cli#async-config) and pass the result to the [`contracts` option](https://wagmi.sh/cli/configuration/options#contracts-optional)
